key/aziot-keys/src/key.rs

```diff
@@ -297,14 +331,35 @@ pub(crate) unsafe fn encrypt(
 
     let cipher = openssl::symm::Cipher::aes_256_gcm();
 
-    let mut tag = vec![0_u8; 16];
-    let ciphertext =
-        openssl::symm::encrypt_aead(cipher, &key, Some(iv), aad, plaintext, &mut tag[..])?;
+    match key {
+        Key::FileSystem(key) => {
+            let mut tag = [0_u8; 16];
+            let mut ciphertext =
+                openssl::symm::encrypt_aead(cipher, &key, Some(iv), aad, plaintext, &mut tag)?;
+
+            ciphertext.insert(0, 0x01);
+            ciphertext.extend_from_slice(&tag);
+            Ok((ciphertext.len(), ciphertext))
```

Comment:
It matches the other functions where that is not true, because it's not true in general. The length is an upper bound for all calls and can be longer than the actual `Vec` for a particular call.